blog/_posts/2016-09-16-juliacon2016.md

```diff
@@ -0,0 +1,154 @@
+Stata Centre, a towering monolith which, the venue for an event that
+people from all over the world came to attend and participate in. I was in town for
+the [third edition of JuliaCon](https://www.youtube.com/watch?v=EZD3Scuv02g&list=PLP8iPy9hna6SQPwZUDtAM59-wPzCPyD_S),
+the annual Julia Conference at MIT.
+
+On the eve of JuliaCon, and a series of workshops were organised on some important
+areas people use Julia for. I was conducting the
+[Parallel Computing workshop](https://www.youtube.com/watch?v=euZkvgx0fG8)
+along with some other members of the JuliaLab. The key idea in our workshop was
+to show users the many different ways of writing and executing parallel code in Julia.
+I was talking about easy GPU computing using my package called ArrayFire and achieving
+acceleration using Julia’s multi-threading.
+
+Day 1 started off with the first keynote speaker in Guy Steele, a stalwart
+in the software industry and an expert in programming language design. He
+[spoke](https://www.youtube.com/watch?v=EZD3Scuv02g) about his adventures designing
+Fortress, a language that was intended to be good at mathematical programming.
+He went through the key design principles and tradeoffs: from the type hierarchy,
+to their model for parallelism (automatic work-stealing), and interesting choices
+(such as non-transitive operator precedence). My colleague Keno Fischer was up next
+with a [tour](https://www.youtube.com/watch?v=e6-hcOHO0tc) of the new Julia Debugger:
+Gallium! Gallium was quite breathtaking in its complexity and versatility, so much so
+that Keno himself uses to debug code in C and C++! A powerful debugger becomes even
+better with GUI-integration, which Mike Innes very usefully
+[pitched in](https://www.youtube.com/watch?v=yDwUL3aRSRc) with during his demo of
+Juno-Gallium integration. Stepping, printing and breakpoints promised a powerful
+package development experience.
+
+The next session was all about data science. Simon Byrne
+[spoke](https://www.youtube.com/watch?v=ScCY_nE0hlU) about the data science ecosystem
+in Julia and future plans. He touched on the
+[famous problem](http://www.johnmyleswhite.com/notebook/2015/11/28/why-julias-dataframes-are-still-slow/)
+with DataFrames, and then laid out a roadmap for the ecosystem. The rest of the
+session featured an interesting demo in
+[music processing](https://www.youtube.com/watch?v=IOVrVOacLP8),
+while Arch Robison [showed](https://www.youtube.com/watch?v=02NkiDoRDCU)
+us how to use Julia as a code generator.
+
+The evening had two sessions in parallel at different rooms. This is a recurrent
+feature of JuliaCon, and it’s always hard to decide which session to attend.
+This time, I chose to attend the sessions on
+[automatic differentation in JuMP](https://www.youtube.com/watch?v=xtfNug-htcs) and
+[forward](https://www.youtube.com/watch?v=r2hhRSHiQwY) differentiation using
+`ForwardDiff.jl`. I didn’t want to miss the
```

Comment:
It would be nice to have links to the packages here in the blog post; links to both `ForwardDiff.jl` and `JuMP`, that is.